File: alg-teachers-support-sorting/2 var/2_task.cpp
#include <iostream>
#include <vector>
#include <map>
using namespace std;
 
void countSort(vector<int> &input)
{
    // создаем пустую карту для хранения частот элементов массива
    map<int, int> freq;
 
    // сохраняем различные значения во входном массиве как ключ и
    // их соответствующие счетчики как значения
    for (int x: input) {
        freq[x]++;
    }
    for (auto &p: freq) {
        cout << p.first << " " << p.second << endl;
    }
 
    // обход карты и перезапись входного массива отсортированными элементами
    // (карта будет повторяться на основе отсортированного порядка ключей)
    int i = 0;
    for (auto &p: freq)
    {
        while (p.second--) {
            input[i++] = p.first;
        }
    }
}
 
int main()
{
    int n;
    int a;
    cin >> n;
    vector<int> input;
    for (int i=0; i<n; i++) {
        cin >> a;
        input.push_back(a);
    }
 
    countSort(input);

    cout << "Answer: ";
    for (int i=0; i<n; i++) {
        if (i == n-1) cout << input[i] << endl;
        else cout << input[i] << " ";
    }
 
    return 0;
}

File: alg-teachers-support-sorting/2 var/2_task.py
def countSort(input):
    # создаем пустой словарь для хранения частот элементов списка
    freq = {}

    # сохраняем различные значения во входном списке как ключ и
    # их соответствующие счетчики как значения
    for x in input:
        freq[x] = freq.get(x, 0) + 1

    for i in range(len(input)-1):
        for j in range(len(input)-i-1):
            if input[j]>input[j+1]:
                input[j], input[j+1] = input[j+1], input[j]
    sorted_input = input
    
    for i in range(len(input)):
        if i!=0:
            if not sorted_input[i]==sorted_input[i-1]:
                print(sorted_input[i][0], freq[sorted_input[i]])
        else:
            print(sorted_input[i][0], freq[sorted_input[i]])


n = int(input())
input_list = []
a_input = input().split()
for i in range(n):
    a = int(a_input[i])
    input_list.append((a, 0))

countSort(input_list)

print("Answer:", ' '.join(str(e[0]) for e in input_list))